File: SuffleTheArray/suffleTheArray.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number[]}
 */
var shuffle = function(nums, n) {
  let new_arr = [];
  let arr_length = (nums.length / 2);
  for (let i = 0; i < n; i++) {
      new_arr.push(nums[i], nums[arr_length + i])
  }
  return new_arr;
};

data = [
  [[2,5,1,3,4,7], 3],
  [[1,2,3,4,4,3,2,1], 4],
  [[1,1,2,2], 2]
] 


data.forEach(element => {
  console.log(shuffle(element[0], element[1]));
});

File: SuffleTheArray/suffleTheArray.py
class Solution:
    def shuffle(self, nums: list[int], n: int) -> list[int]:
        new_arr = []
        for i in range(n):
            new_arr.append(nums[i])
            new_arr.append(nums[i + n])
        return new_arr
    


nums = [1,2,3,4,5,6,7,8]
n = 4
sol = Solution()

print(sol.shuffle(nums, n))